client/src/components/All_Modules_Page/AllModulesPage.jsx

```diff
@@ -1,64 +1,43 @@
+import React, { useState, useEffect } from "react";
 import "./AllModulesPage.css";
 import CardForTopic from "./Card_For_Topic/Card_for_topic";
-const fakedata = [
-  {
-    module: "HTML||CSS",
-    topics: [
-      {
-        nameOfTopic: "Tags",
-        details:
-          "If you're a Dropbox Pro or Dropbox for Business user, you can control who accesses your links and for how long by setting passwords and expirations for them. Alternatively, you can use view-only permissions for shared folders. Links are ideal for replacing email attachments, whereas view-only permissions for shared folders are useful for keeping collaborators continuously up to date without giving them the ability to change files..",
-      },
-      {
-        nameOfTopic: "Styling",
-        details: "Array of ayyj ooen ooog hynslki oenvttb ioo, ppp.",
-      },
-    ],
-  },
-  {
-    module: "JavaScript 1",
-    topics: [
-      {
-        nameOfTopic: "Array",
-        details: "Array of ayyj ooen ooog hynslki oenvttb ioo, ppp.",
-      },
-      {
-        nameOfTopic: "Object",
-        details:
-          "If you're a Dropbox Pro or Dropbox for Business user, you can control who accesses your links and for how long by setting passwords and expirations for them. Alternatively, you can use view-only permissions for shared folders. Links are ideal for replacing email attachments, whereas view-only permissions for shared folders are useful for keeping collaborators continuously up to date without giving them the ability to change files..",
-      },
-    ],
-  },
-  {
-    module: "React",
-    topics: [
-      {
-        nameOfTopic: "Components",
-        details:
-          "If you're a Dropbox Pro or Dropbox for Business user, you can control who accesses your links and for how long by setting passwords and expirations for them. Alternatively, you can use view-only permissions for shared folders. Links are ideal for replacing email attachments, whereas view-only permissions for shared folders are useful for keeping collaborators continuously up to date without giving them the ability to change files..",
-      },
-      {
-        nameOfTopic: "Imports",
-        details:
-          "If you're a Dropbox Pro or Dropbox for Business user, you can control who accesses your links and for how long by setting passwords and expirations for them. Alternatively, you can use view-only permissions for shared folders. Links are ideal for replacing email attachments, whereas view-only permissions for shared folders are useful for keeping collaborators continuously up to date without giving them the ability to change files..",
-      },
-    ],
-  },
-];
-export default function AllModulesPage() {
+
+export default function AllModulesPage({ refreshmodalData }) {
+  const [modalData, setModalData] = useState([]);
+
+  const fetchData = async () => {
+    try {
+      const response = await fetch(
+        `${process.env.REACT_APP_DB_URL}/allModulesPage`
```

Comment:
Hey Kristian, there is something important I need to let you know. The environment variable that you are trying to access is only available on your local laptop. It won't show up if we merge it. Please use the backend URL for now instead. We will work on making this variable available on our deploy website soon.

process.env.REACT_APP_DB_URL <<< this variable